### PR TITLE
Migrate to Only Using Core Modeling API

### LINKED ISSUE #535: Only Use Modeling#updateProperties and Modeling#updateModdleProperties in Properties Panel
* only use Modeling#updateProperties and Modeling#updateModdleProperties in properties panel
* remove additional commands from properties panel (e.g. properties-panel.update-businessobject)

---

Child of https://github.com/bpmn-io/internal-docs/issues/358

### PR DESCRIPTION
**Pull request by philippfromme**
# Changes

* only use `Modeling#updateProperties` and `Modeling#updateModdleProperties`
* remove `properties-panel.update-businessobject` and `properties-panel.update-businessobject-list` commands
* do not remove empty extension elements

Closes #535

### CODE REVIEW

File: src/provider/bpmn/properties/CompensationProps.js [hunk 1]
```diff
@@ -69,11 +69,11 @@ function WaitForCompletion(props) {
   };
 
   const setValue = (value) => {
-    commandStack.execute('properties-panel.update-businessobject', {
-      element: element,
-      businessObject: compensateEventDefinition,
+    commandStack.execute('element.updateModdleProperties', {
```
Comment (barmac) [inline src/provider/bpmn/properties/CompensationProps.js:72]: How about if we use `modeling.updateModdleProperties` explicitly?

File: src/provider/camunda-platform/properties/ExtensionPropertiesProps.js [hunk 2]
```diff
@@ -63,27 +65,49 @@ function removeFactory({ commandStack, element, property }) {
...
+            moddleElement: businessObject,
+            properties: {
+              extensionElements: undefined
+            }
+          }
+        });
+      }
     }
 
     commandStack.execute('properties-panel.multi-command-executor', commands);
```
Comment (barmac) [inline src/provider/camunda-platform/properties/ExtensionPropertiesProps.js:113]: Hmm if we really migrate to only use core modeling API, we should also get rid of the multi command thing.

File: src/provider/camunda-platform/properties/ExtensionPropertiesProps.js [hunk 3]
```diff
@@ -63,27 +65,49 @@ function removeFactory({ commandStack, element, property }) {
...
       });
+
+      if (!values.length) {
+        commands.push({
+          cmd: 'element.updateModdleProperties',
+          context: {
+            element,
+            moddleElement: businessObject,
+            properties: {
+              extensionElements: undefined
```
Comment (barmac) [inline src/provider/camunda-platform/properties/ExtensionPropertiesProps.js:106]: Can we *not* do that? Note that this behaviour is a reported bug: https://github.com/bpmn-io/bpmn-js-properties-panel/issues/499

File: src/provider/camunda-platform/properties/InMappingPropagationProps.js [hunk 4]
```diff
@@ -83,66 +85,25 @@ function PropagateAll(props) {
...
-    }
-
-    // (2) Add camunda:In mapping
     const mapping = createElement(
       'camunda:In',
       {
         variables: 'all'
       },
-      parent,
+      null,
```
Comment (barmac) [inline src/provider/camunda-platform/properties/InMappingPropagationProps.js:95]: Is the parent set somewhere downstream?

Comment (barmac) [reply src/provider/camunda-platform/properties/InMappingPropagationProps.js:95]: Nevermind, I can see it is set in the util.

**Comment by barmac** [comment]
I left a couple of comments which need to be addressed before we merge this. 🏅 for a tremendous work done in this PR.

File: src/provider/bpmn/properties/CompensationProps.js (hunk 1 above)
Comment (philippfromme) [reply src/provider/bpmn/properties/CompensationProps.js:72]: We could but I wanted to keep the changes as simple as possible. Also in the case of multiple commands, we have to use the command stack as there is no modeling API.

Comment (barmac) [reply src/provider/bpmn/properties/CompensationProps.js:72]: OK, I guess we can also do that for the spring cleaning.

File: src/provider/camunda-platform/properties/ExtensionPropertiesProps.js (hunk 2 above)
Comment (philippfromme) [reply src/provider/camunda-platform/properties/ExtensionPropertiesProps.js:113]: Same here. I wanted to keep the changes simple. 🤡

Comment (barmac) [reply src/provider/camunda-platform/properties/ExtensionPropertiesProps.js:113]: Which is fine.

File: src/provider/camunda-platform/properties/ExtensionPropertiesProps.js (hunk 3 above)
Comment (philippfromme) [reply src/provider/camunda-platform/properties/ExtensionPropertiesProps.js:106]: In that case what should we do? Remove all instances of _remove empty extension elements_?

Comment (barmac) [reply src/provider/camunda-platform/properties/ExtensionPropertiesProps.js:106]: Yeah. When I remove an extension, I remove the item from extension elements values and only that. There can be a separate behaviour which could perform the cleanup.

Comment (philippfromme) [reply src/provider/camunda-platform/properties/ExtensionPropertiesProps.js]: I've removed all instances of removing what we assume to be empty extension elements (cf. https://github.com/bpmn-io/bpmn-js-properties-panel/pull/536/commits/cbcf363fa4786b85341d1cc592f64dd8a2d6ff55).

**Review by barmac — APPROVED** [review]
❤️

**Comment by philippfromme** [comment]
🚀

**Pull request merged**